charts/plugin-barman-cloud/templates/secret.yaml

```
@@ -0,0 +1,26 @@
+#
+# Copyright The CloudNativePG Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+---
+apiVersion: v1
+data:
+  SIDECAR_IMAGE: |
+    Z2hjci5pby9jbG91ZG5hdGl2ZS1wZy9wbHVnaW4tYmFybWFuLWNsb3VkLXNpZGVjYXI6dj
+    AuNC4w
+kind: Secret
+metadata:
+  name: plugin-barman-cloud-m76km67hd7
+  namespace: {{ include "plugin-barman-cloud.namespace" . }}
+type: Opaque
```

This contains the sidecar image name and version that should be injected by the controller. I think this can be an imageRef value that is templated into a configmap.

```suggestion
apiVersion: v1
data:
  SIDECAR_IMAGE: |
    "{{ .Values.sidecarImage.registry }}{{ .Values.sidecarImage.repository }}:{{ .Values.sidecarImage.tag | default .Chart.AppVersion }}"
kind: ConfigMap
metadata:
  name: plugin-barman-cloud-config
  namespace: {{ include "plugin-barman-cloud.namespace" . }}

```

---

I also renamed the file to match its new kind